Rewrite Cassandra tests using Alia

craigpastro opened the pull request:
Tried to clean up some of the unused imports as well as some formatting (formatting is using the default values of Cursive which seem to be pretty standard... probably?)

All the Cassandra tests work as before, same as the ScalarDB tests.

Commit:
Rewrite Cassandra tests using Alia

Tried to clean up some of the unused imports as well as some formatting (formatting is using the default values of Cursive which seem to be pretty standard...?)

All the Cassandra tests work as before, same as the ScalarDB tests.

Commit:
formatting

feeblefakie commented on cassandra/src/cassandra/lwt.clj:
```diff
@@ -1,139 +1,93 @@
+                               :value)]
+                (assoc op :type :ok, :value value)))
+
+      (catch ExceptionInfo e
```
As we discussed in person, it seems good to catch the same exceptions in the same way as before, otherwise it works differently from the previous one.
As for the scalar db tests, it won't be a big problem, but for C* tests, I think it makes a difference.

craigpastro replied:
Very good point. I finally figured out how to do it, so fixed! Thank you.

feeblefakie reviewed (changes requested):
Overall looking good but I have some concerns in exception handling in the new code as we discussed in person.
It might be good to ask Alia driver developer how to rewrite the cassaforte's exception handling.

Commit:
Branch based on the exception caught

I finally figured out how to check the type of exception thrown by the Datastax driver, so now I can branch on the type of exception. The tests should now work as with the old driver (hopefully better?).

yito88 commented on cassandra/src/cassandra/collections/map.clj:
```diff
@@ -1,101 +1,82 @@
+
+      (catch ExceptionInfo e
+        (let [e (class (:exception (ex-data e)))]
+          (cond
```
How about this?

```clojure
          (condp = e
            ReadTimeoutException (assoc op :type :fail, :error :read-timed-out)
            WriteTimeoutException (assoc op :type :fail, :error :write-timed-out)
            ...
```

craigpastro replied:
Nice! Thank you. That cleans it up. I was always a bit confused on how to use `condp` because I think all the examples I had seen used `<` or `>`, which made it a bit confusing to parse where the argument was going.

yito88 reviewed (commented):
Looks good! Thanks!
I left a comment.

Commit:
Use `condp` instead of `cond` in exception handling

yito88 reviewed (approved):
LGTM!

feeblefakie reviewed (approved):
LGTM !

yito88 reviewed (changes requested):
Sorry. In my environment, `lwt` test failed due to inconsistencies.
Can you try it?

```
lein run test --test lwt
```

Commit:
WriteTimeouts should be infos and not fails

This fixes a bug that writes in progress were not accounted for in the analysis.

siyopao commented:
@yito88 I think the latest commit fixes the problem with inconsistencies. Thank you for noticing! It was definitely a serious bug.

feeblefakie reviewed (approved):
LGTM!

yito88 commented on cassandra/src/cassandra/lwt.clj:
```diff
@@ -1,139 +1,108 @@
+                                        (update :lwt
+                                                (set-columns {:value new})
+                                                (where [[:id 0]])
+                                                (only-if [[:value old]])))]
```
Is it ok not to specify the consistency level?

I got a log about a UPDATE statement with `CL.LOCAL_ONE`.
The test with a nemesis might fail.

craigpastro replied:
Thank you for your question. I'm doing some investigation now about this matter, and I'll get back to you when I have some definitive answers.

By the way, I did try to run the `lwt` test with all sorts of nemeses and never got any inconsistencies... but who knows.

yito88 commented on cassandra/src/cassandra/lwt.clj:
```diff
@@ -1,139 +1,108 @@
+                     result (alia/execute conn (update :lwt
+                                                       (set-columns {:value v})
+                                                       (only-if [[:in :value (range 5)]])
+                                                       (where [[= :id 0]])))]
```
ditto

yito88 commented on cassandra/src/cassandra/lwt.clj:
```diff
@@ -1,139 +1,108 @@
+                   (let [result' (alia/execute conn (insert :lwt
+                                                            (values [[:id 0]
+                                                                     [:value v]])
+                                                            (if-exists false)))]
```
ditto

yito88 reviewed (changes requested):
I left comments. Can you re-check the consistency level for all writes not only in LWT test but also in other tests?

yito88 reviewed (approved):
LGTM

Sorry. As we discussed, the default `serial_consistency_level` is SERIAL.
The writes work as expected.

siyopao commented:
@yito88 No no, thank you for the comment. I needed to learn a lot about LWTs and Cassandra in general.
